src/components/SearchModal/CurrencySearch.tsx

```diff
@@ -325,7 +333,35 @@ export function CurrencySearch({
       const nextPage = (page ?? pageCount) + 1
       let tokens: WrappedTokenInfo[] = []
       if (debouncedQuery) {
-        tokens = await fetchTokens(debouncedQuery, nextPage, chainId)
+        const promiseResult = await Promise.allSettled([
```

Comment:
SHould we use Promis.all() instead of allSettled? Some old browser does not support allSettled

Reply:
Promise.all doesn't fit here. If the RPC call fails, Promise.all will reject immediately. 
I updated this a bit.
RPC request will be made if fetchTokens fails or returns empty array.